File: backend/app/models/consent.py
from sqlalchemy import Column, String, Boolean, ForeignKey, Text, DateTime
from sqlalchemy.orm import relationship
from app.core.database import Base
from app.models.base import UUIDMixin


class Consent(Base, UUIDMixin):
    __tablename__ = "consents"

    review_id = Column(String, ForeignKey("reviews.id"), unique=True, nullable=False)
    obtained = Column(Boolean, default=False)
    obtained_at = Column(DateTime, nullable=True)
    notes = Column(Text, nullable=True)

    # Relationships
    review = relationship("Review", back_populates="consent")

File: frontend/src/api/consent.ts
import apiClient from './client';

export interface Consent {
  id: string;
  review_id: string;
  obtained: boolean;
  obtained_at: string | null;
  notes: string | null;
}

export interface ConsentUpdate {
  obtained: boolean;
  notes?: string;
}

export const consentApi = {
  getConsent: async (reviewId: string): Promise<Consent> => {
    const response = await apiClient.get<Consent>(`/reviews/${reviewId}/consent`);
    return response.data;
  },

  updateConsent: async (reviewId: string, data: ConsentUpdate): Promise<Consent> => {
    const response = await apiClient.put<Consent>(`/reviews/${reviewId}/consent`, data);
    return response.data;
  },
};
